File: insertion_sort/insertion_sort.cpp
//insertion sort
#include <iostream>

void insertion_sort(){
    int A[10] = {10,9,8,7,6,5,4,3,2,1};
	std::cout<<"A ";
	for (int j = 0; j<10; j++){
	    std::cout<< A[j]<<" ";		
	}

    for (int i=0; i<10; i++){
        int temp = i;
        while (temp > 0 and A[temp-1]>A[temp]){
            int temp_b = A[temp];
            A[temp] = A[temp-1];
            A[temp-1] = temp_b;
            temp = temp - 1;
        }
    }
	std::cout<< '\n' <<"A ";
	for (int k = 0; k<10; k++){
	    std::cout<< A[k] <<" ";		
	}
	std::cout<<std::endl;
}


int main(){
    insertion_sort();
    return 0;
}

File: insertion_sort/insertion_sort.py
#!/usr/bin/python

#insertion sort

def insertion_sort(A):
    for i in range(0,len(A)):
        temp = i
        while (temp > 0 and A[temp-1]>A[temp]):
            temp_b = A[temp]
            A[temp] = A[temp-1]
            A[temp-1] = temp_b
            temp = temp - 1
    return A

if __name__ == '__main__':
    A = [1,2,3,4,5,6,7,8,9,10]
    A = [10,9,8,7,6,5,4,3,2,1]
    print "A= ", A
    insertion_sort(A)
    print "A= ", A
